application/src/main/java/module-info.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2011-2025 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+module io.vertx.launcher.application {
+  requires info.picocli;
+  requires io.vertx.core;
+  requires io.vertx.core.logging;
+
+  // Annotation processing
+  requires static io.vertx.docgen;
+
+  uses io.vertx.core.spi.VertxServiceProvider;
+  opens io.vertx.launcher.application.impl to info.picocli;
```

Comment:
interesting

Reply:
This is required for Picocli to access the fields annotated in our command implementation.